Spec/RealtimeClientConnection.swift

```diff
@@ -3468,6 +3468,88 @@ class RealtimeClientConnection: QuickSpec {
 
             }
 
+            // RTN24
+            it("the client may receive a CONNECTED ProtocolMessage from Ably at any point and should emit an UPDATE event") {
+                let options = AblyTests.commonAppSetup()
+                options.useTokenAuth = true
+                let client = ARTRealtime(options: options)
+                defer { client.dispose(); client.close() }
+
+                waitUntil(timeout: testTimeout) { done in
+                    client.connection.once(.Connected) { stateChange in
+                        expect(stateChange?.reason).to(beNil())
+                        done()
+                    }
+                }
+
+                client.connection.once(.Connected) { stateChange in
+                    fail("Should not emit a Connected state")
+                }
+
+                waitUntil(timeout: testTimeout) { done in
+                    client.connection.once(.Update) { stateChange in
+                        guard let stateChange = stateChange else {
+                            fail("ConnectionStateChange is nil"); done(); return
+                        }
+                        expect(stateChange.reason).to(beNil())
+                        expect(client.connection.state).to(equal(ARTRealtimeConnectionState.Connected))
+                        expect(stateChange.current).to(equal(ARTRealtimeConnectionState.Connected))
+                        expect(stateChange.current).to(equal(stateChange.previous))
+                        done()
+                    }
+
+                    client.auth.authorize(nil, options: nil) { tokenDetails, error in
```

Comment:
I am not sure that an authorize will always respond with a `CONNECTED` message to be honest. Perhaps what you should do instead, so that the test is closer to the behaviour we expect is:

* Initially connect with token auth callback but no `clientId`, setting a lot TTL of say 35 seconds (I think if > 30s the server will send a AUTH protocol message at 30s before expiry)
* Ensure you are connected
* Wait for the auth callback to be called again but this time auth with a clientId.  We can then be 100% sure the CONNECTED message with new details is expected
* Check that the UPDATE is emitted as you are now and the clientId has changed